compose/foundation/foundation/src/uikitMain/kotlin/androidx/compose/foundation/gestures/UIKitTapGestureDetector.kt

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.foundation.gestures
+
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.input.pointer.AwaitPointerEventScope
+import androidx.compose.ui.input.pointer.PointerEvent
+import androidx.compose.ui.input.pointer.PointerEventPass
+import androidx.compose.ui.input.pointer.PointerInputChange
+import androidx.compose.ui.input.pointer.PointerInputScope
+import androidx.compose.ui.input.pointer.PointerType
+import androidx.compose.ui.input.pointer.changedToDown
+import androidx.compose.ui.platform.ViewConfiguration
+import androidx.compose.ui.util.fastAll
+
+/**
+ * Detects repeating tap gestures on the given PointerInputScope.
+ *
+ * **Important**: for every tap sequence corresponding callback will be called.
+ *
+ * For example, if there is triple tap gesture, the call stack will look as follows: onTap, onDoubleTap, onTripleTap
+ *
+ * If there is a double tap gesture, the call stack will look as follows: onTap, onDoubleTap
+ *
+ * The order of callback execution is guaranteed.
+ *
+ * @param onTap  A callback function that will be called when a single tap gesture is detected. It will receive the position (Offset) where the tap occurred.
+ * @param onDoubleTap  A callback function that will be called when a double tap gesture is detected. It will receive the position (Offset) where the tap occurred.
+ * @param onTripleTap  A callback function that will be called when a triple tap gesture is detected. It will receive the position (Offset) where the tap occurred.
+ */
+internal suspend fun PointerInputScope.detectRepeatingTapGestures(
+    onTap: ((Offset) -> Unit)? = null,
+    onDoubleTap: ((Offset) -> Unit)? = null,
+    onTripleTap: ((Offset) -> Unit)? = null,
+) {
+    awaitEachGesture {
+        val touchesCounter = TouchCounter(viewConfiguration)
+        while (true) {
+            val down = awaitTouchEventDown()
```

Comment:
```suggestion
            val down = awaitFirstDown()
```

`awaitFirstDown` works for Touch and for Mouse the same way as `awaitTouchEventDown` and `awaitMouseEventDown` do.

Reply:
We need to change `fun update(event: PointerEvent)` to `fun update(event: PointerInputChange)` after that.

Reply:
done